src/Analysis/Ast/Impl/Linting/UndefinedVariables/UndefinedVariablesWalker.cs

```diff
@@ -53,11 +52,9 @@ public UndefinedVariablesWalker(IDocumentAnalysis analysis, IServiceContainer se
                         augs.Right?.Walk(new ExpressionWalker(this));
                         break;
                     case AssignmentStatement asst:
-                        _suppressDiagnostics = true;
-                        foreach (var lhs in asst.Left ?? Enumerable.Empty<Expression>()) {
+                        foreach (var lhs in asst.Left.MaybeEnumerate().Where(x => !(x is NameExpression))) {
```

Comment:
What affect does this have for an assignment like:

```python
def foo():
    return 1, 2

x = 1234
x, y = foo()
```

Where `y` has not been referenced but is being walked as a name expression?

I'm thinking that the original solution was "more correct" in that it's trying to avoid lvars being linted, but needed an extra case where it stored `_suppressDiagnostics` and made it false before doing the first member access (to check the first thing), and then restored it.